cpp/dolfinx/fem/ElementDofLayout.cpp

```diff
@@ -75,6 +75,16 @@ ElementDofLayout::ElementDofLayout(
       _num_dofs += entity_dofs_[dim][entity_index].size();
     }
   }
+
+  // Assert that base_permutations is the correct shape
+#ifdef DEBUG
```

Comment:
This is quite a cheap, so I would suggest doing it in all cases and throwing a `runtime_error` if the sizes are wrong.